### PR TITLE
Refactor out an old cooldown system in assemblies

### PR DESCRIPTION
**Pull request by optimumtact**
You can do this a lot more efficiently with world time cooldowns,
avoiding the need for spawn counters

I want to look into some other items that do this (the door controllers)

### CODE REVIEW

File: code/modules/assembly/doorcontrol.dm [hunk 1]
```diff
@@ -6,6 +6,7 @@
 	attachable = 1
 	var/id = null
 	var/can_change_id = 0
+	var/cooldown = 0//Door cooldowns
```
Comment (optimumtact) [inline code/modules/assembly/doorcontrol.dm]: TODO: investigate this file

File: code/modules/assembly/infrared.dm [hunk 2]
```diff
@@ -92,17 +92,15 @@
...
 	if(!holder)
 		return 0
-//	setDir(holder.dir)
 	qdel(first)
 	return 1
 
 /obj/item/device/assembly/infra/proc/trigger_beam()
-	if((!secured)||(!on)||(cooldown > 0))
-		return 0
+	if((!secured) || (!on)|| (next_activate > world.time))
```
Comment (ChangelingRain) [inline code/modules/assembly/infrared.dm]: why do these all have parentheses around them, anyway?

File: code/modules/assembly/proximity.dm [hunk 3]
```diff
@@ -59,12 +59,11 @@
 
 
 /obj/item/device/assembly/prox_sensor/sense()
-	if((!secured)||(cooldown > 0))
+	if((!secured) || (next_activate > world.time))
```
Comment (ChangelingRain) [inline code/modules/assembly/proximity.dm]: parenthememes

File: code/modules/assembly/timer.dm [hunk 4]
```diff
@@ -42,15 +42,12 @@
 
 
 /obj/item/device/assembly/timer/proc/timer_end()
-	if((!secured)||(cooldown > 0))
-		return 0
+	if((!secured)||(next_activate > world.time))
```
Comment (ChangelingRain) [inline code/modules/assembly/timer.dm]: I charge you with fixing these thoroughly unneeded parentheses

File: code/modules/assembly/doorcontrol.dm (hunk 1 above)
Comment (optimumtact) [inline code/modules/assembly/doorcontrol.dm]: one day

**Comment by optimumtact** [comment]
@tgstation/commit-access

**Pull request merged**